File: Week02/LeetCode-0219.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        """
        O(N^2) algorithm: Gets Time Limit Exceeded
        """
        for i in range(len(nums)-1):
            for j in range(i+1, min(i+k+1, len(nums))):
                if nums[i] == nums[j]:
                    return True        
        return False
        

File: Week02/LeetCode-0219.cpp
#include <algorithm>
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        /// O(N^2) algorithm: Gets Time Limit Exceeded
        for (int i=0; i<nums.size()-1; i++){
            for (int j=i+1; j<std::min(i+k+1, (int)nums.size()); j++)
                if (nums[i] == nums[j])
                    return true;
        } // end-containsNearbyDuplicate
        
        return false;
    } // end-containsNearbyDuplicate
};